### DISCUSSION TITLE
"null" type for not required field

### DISCUSSION THREAD

**Discussion by feloy** [swaggest/jsonschema-go#86] [Q&A]
Hi,

I'm starting using jsonschema-go to define a JSON schema. thanks for this work :)

I can see that, when I declare a field as a pointer into a struct, to make it not required, the type "null" is added to the list of types for this field.

```
type MyObject struct {
	_ struct{} `additionalProperties:"false"`
	OptionalField *string `json:"optionalField"`
}
```

```
"optionalField": {
"type": [
      "null",
      "string"
     ]
  }
```

I would expect that the type is only "string" here, as the null value is covered by the not required capability.

**Comment by vearutop** [comment]
Hello, `required` keyword addresses the absence of a property, not its value. For example, for a schema

```json
{"required":["foo"]}
```
values `{"foo":null}`, `{"foo":123}` would be valid,
but values `{}`, `{"bar":null}` would be invalid.
